src/pages/StudentDashboardPage/AssignmentSubmission/AssignmentSubmission.js

```diff
@@ -110,25 +157,74 @@ function AssignmentSubmission() {
       <Box className={classes.submissionsection}>
         <Typography variant="h5">My Submission</Typography>
         <IconButton>
-          <ControlPointOutlinedIcon />
+          <ControlPointOutlinedIcon onClick={handleOpen} />
+          <Modal
+            aria-labelledby="transition-modal-title"
+            className={classes.modal}
+            open={open}
+            onClose={handleClose}
+            closeAfterTransition
+            BackdropComponent={Backdrop}
+            BackdropProps={{
+              timeout: 500,
+            }}
+          >
+            <Fade in={open}>
+              <div className={classes.paper}>
+                <Typography style={{ textAlign: "center" }}>Upload Assignment</Typography>
+                <Divider />
+                <br />
+                <div style={{ textAlign: "center" }}>
+                  <Button
+                    variant="contained"
+                    color="default"
+                    component="span"
+                    startIcon={<AddIcon />}
+                    onClick={onChange}
+                  >
+                    Add
+                  </Button>
+                </div>
+              </div>
+            </Fade>
+          </Modal>
         </IconButton>
       </Box>
       <Box>
-        {[1, 2, 3, 4].map((items, index) => (
-          <List key={index} component="nav" className={classes.list} aria-label="contacts">
-            <ListItem>
-              <ListItemIcon>
-                <Assignment />
-              </ListItemIcon>
-              <ListItemText primary="Assignment1.pdf" />
-              <ListItemIcon>
-                <IconButton>
-                  <Delete />
-                </IconButton>
-              </ListItemIcon>
-            </ListItem>
-          </List>
-        ))}
+        {files
+          ? files.map((file, index) => (
+              <List key={index} component="nav" className={classes.list} aria-label="contacts">
+                <Link href={file.source} color="inherit" target="_blank" underline="none">
+                  <ListItem>
+                    <ListItemIcon>
+                      <Assignment />
+                    </ListItemIcon>
+                    <ListItemText primary={file.name} />
+                    <span> Size: {file.size} bytes </span>
+                    <ListItemIcon>
+                      <IconButton>
+                        <Delete />
+                      </IconButton>
+                    </ListItemIcon>
+                  </ListItem>
+                </Link>
+              </List>
+            ))
```

Comment:
I would suggest making a different component of the assignment list and pass props to it!